### PR TITLE
fix moving and editing of enrollments

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes a long lost work item [PLAT-1386](https://codedotorg.atlassian.net/browse/PLAT-1386) to make it so that you can actually select enrollments and move or rename them.

### problem 1

 avoiding this annoying workaround:

https://user-images.githubusercontent.com/8001765/173941315-ad45b20b-c5a5-4376-9eaf-e2c8f745edf7.mov

now it works like this:

https://user-images.githubusercontent.com/8001765/173941413-4a4a63e7-ceda-4726-a668-748bb280b218.mov

The problem is that the original code was not following the [contract for set state](https://reactjs.org/docs/react-component.html#setstate), and was breaking the rule which says that you should never modify react state directly.

### problem 2

also fixes this problem:

https://user-images.githubusercontent.com/8001765/173957967-47fa402e-0318-4671-8018-32118f88dc9a.mov

which had this annoying workaround, which avoids using the mouse by typing: tab, tab, [number], tab, enter: 

https://user-images.githubusercontent.com/8001765/173957584-2f1a1ec5-165e-431f-95cc-81e74de5fb17.mov

a similar problem / workaround existed for editing name.

now it works like this:

https://user-images.githubusercontent.com/8001765/173957599-bdbda68f-33fc-4b0d-98c6-ed0fb553cf6c.mov

this one is a bit more mysterious. I noticed that `handleClickChangeEnrollments` was getting called with `event.target.value == ''` when I click as shown in the first screenshot. I didn't see anything wrong with how `handleClickChangeEnrollments` was configured, so I opted to solve this by locking down the implementation of `handleClickChangeEnrollments` so that it could only be used to open the dialog, never to close it.

## Testing story

* manual verification shown in screenshots
* removed workarounds from existing unit tests

### CODE REVIEW

**Comment by davidsbailey** [comment]
hi reviewers, apologies for adding more functionality after requesting review. the PR description has been updated to match, and now it should be ready to review again.

File: apps/src/code-studio/pd/workshop_dashboard/EnrollmentsPanel.jsx [hunk 1]
```diff
@@ -160,12 +166,13 @@ export default class EnrollmentsPanel extends React.Component {
       });
     } else {
       this.setState(state => {
-        state.selectedEnrollments.push({
+        const selectedEnrollments = state.selectedEnrollments.concat({
```
Comment (megcrenshaw) [inline apps/src/code-studio/pd/workshop_dashboard/EnrollmentsPanel.jsx:169]: Wow, I've never seen this way of modifying state!

**Review by megcrenshaw — APPROVED** [review]
Thanks for cleaning this up and saving us more requests like this!

File: apps/src/code-studio/pd/workshop_dashboard/EnrollmentsPanel.jsx (hunk 1 above)
Comment (davidsbailey) [reply apps/src/code-studio/pd/workshop_dashboard/EnrollmentsPanel.jsx:169]: yes, it is a bit overkill! I think this way only exists for when you need to do something that's more complicated than what's being done here, but the change I'm making didn't seem big enough to be worth reimplementing it.

**Pull request merged**